internal/app/accrualdaemon/accrual_daemon.go

```diff
@@ -0,0 +1,154 @@
+package accrualdaemon
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go.uber.org/zap"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+	"yandex_gophermart/pkg/entities"
+	gophermart_errors "yandex_gophermart/pkg/errors"
+)
+
+//
+//const (
+//	orderStatusRegistered = "REGISTERED"
+//	orderStatusInvalid    = "INVALID"
+//	orderStatusProcessing = "PROCESSING"
+//	orderStatusProcessed  = "PROCESSED"
+//)
+
+type UnfinishedOrdersStorageInt interface {
+	UpdateOrder(orderData entities.OrderData, ctx context.Context) error
+	GetUnfinishedOrdersList(ctx context.Context) ([]entities.OrderData, error)
+	AddToBalance(userID int, amount float64, ctx context.Context) error
+}
+
+type respData struct {
+	Order   string  `json:"order"`
+	Status  string  `json:"status"`
+	Accrual float64 `json:"accrual"`
+}
+
+func AccrualCheckDaemon(ctx context.Context, logger *zap.SugaredLogger, storage UnfinishedOrdersStorageInt, accrualSystemAddress string) {
```

Comment:
```suggestion
func AccrualCheckDaemon(ctx context.Context, logger *zap.SugaredLogger, storage UnfinishedOrdersStorageInt, accrualSystemAddress string, wg &sync.WaitGroup) {
    defer wg.Done()
```

При шатдауне желательно дождаться завершения работы accrual-интерфейса, чтобы не потерять данные или не попасть на панику.

Reply:
fix